### ISSUE TITLE
npm packages backup: Doesn't handle scopes, fails with unmet peer dependencies

### ISSUE THREAD

**Issue by jnv** [alichtman/shallow-backup#226]
Hi, first of all, thanks for this program, I have switched to shallow-backup from Mackup because it handles only important stuff and helps me migrate configuration between Mac and Linux.

That being said, shallow-backup often fails to backup globally installed npm packages for me and the backed up list is not correct.

First, `npm ls` command can often exit with error due to unmet peer dependecies. It writes out packages into stdout correctly but then it writes error into stderr like:
```
npm ERR! peer dep missing: eslint@^5.0.0, required by @typescript-eslint/eslint-plugin@1.12.0
npm ERR! peer dep missing: eslint@^5.0.0, required by @typescript-eslint/parser@1.12.0
```
and exits with `1`. This causes shallow-backup to fail and delete the package list.

The second issue is with the actual package list for packages under `@scope`.  shallow-backup uses only the last part of path after slash:
https://github.com/alichtman/shallow-backup/blob/992a53ee90ef7f919950e73ebeb9dcc654be746b/shallow_backup/backup.py#L137
Therefore for packages like `@babel/cli` only `cli` gets saved.

I think both of these issues can be solved by using JSON output, i.e.:

```
npm ls --global --json --depth=0
```

This way npm ignores (most) unmet peer dependencies so it doesn't exit with error code and can be easily parsed for correct package name, example output:
```json
{
  "dependencies": {
    "@babel/cli": {
      "version": "7.5.0",
      "from": "@babel/cli",
      "resolved": "https://registry.npmjs.org/@babel/cli/-/cli-7.5.0.tgz"
    },
    "npm": {
      "version": "6.10.1",
      "from": "npm",
      "resolved": "https://registry.npmjs.org/npm/-/npm-6.10.1.tgz"
    },
    "npm-check-updates": {
      "version": "3.1.18",
      "from": "npm-check-updates",
      "resolved": "https://registry.npmjs.org/npm-check-updates/-/npm-check-updates-3.1.18.tgz"
    },
    "tldr": {
      "version": "3.2.7",
      "from": "tldr",
      "resolved": "https://registry.npmjs.org/tldr/-/tldr-3.2.7.tgz"
    }
  }
}
```

Would you be interested in changing npm backup behavior to use JSON output? I could try to make a PR for this.

**Comment by alichtman** [comment]
Hey there! I’m glad to hear that you’re getting value from this!

Your solution sounds good to me, and I’d gladly accept a PR.

**Comment by jnv** [comment]
Looks like this was fixed in #315. Thanks @nick-The-One!
